File: src/main/java/net/nodus/core/site/dto/SiteResponse.java
package net.nodus.core.site.dto;

import net.nodus.database.site.entity.Site;

import java.util.Objects;

public record SiteResponse(
    String id,
    String name,
    String domain, String url, String siteKey) {

    public SiteResponse(String id, String name, String domain, String url) {
        this(id, name, domain, url, null);
    }

    public static SiteResponse from(Site site) {
        return new SiteResponse(
            Objects.requireNonNull(site.getId()),
            site.getName(),
            site.getDomain(),
            site.getUrl()
        );
    }
}

File: src/main/kotlin/net/nodus/site/dto/SiteResponse.kt
package net.nodus.site.dto

import net.nodus.site.SiteKeyStatus
import net.nodus.site.entity.Site
import net.nodus.site.entity.SiteKey

data class SiteResponse(
    val id: String,
    val name: String,
    val domain: String?,
    val url: String?,
    val siteKey: String? = null
)

data class SiteKeyResponse(
    val keyId: String,
    val keyPrefix: String,
    val siteKey: String? = null,
    val status: SiteKeyStatus,
)

data class SiteCreateResult(
    val site: Site,
    val issuedKey: IssuedSiteKey,
)

data class IssuedSiteKey(
    val siteKey: SiteKey,
    val rawKey: String,
)

fun Site.toResponse(): SiteResponse =
    SiteResponse(
        id = requireNotNull(id),
        name = name,
        domain = domain,
        url = url,
    )

fun SiteKey.toResponse(): SiteKeyResponse =
    SiteKeyResponse(
        keyId = requireNotNull(id),
        keyPrefix = keyPrefix,
        status = status,
    )